TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -115,9 +120,76 @@ public override bool Visit(DataDefinition dataDefinition)
                     "The variable '" + dataDefinition.Name + "' can only be of level 01 or 77");
             }
 
+
+            //get parent of node - DataDescription entries and Data Redefines entries cannot have childrens
+            dynamic nodeParentDataDescription = dataDefinition.Parent?.CodeElement as DataDescriptionEntry;
+            if (nodeParentDataDescription == null)
+            {
+                nodeParentDataDescription = dataDefinition.Parent?.CodeElement as DataRedefinesEntry;
+            }
+
+            //return if node is level 01
+            if (nodeParentDataDescription == null)
+            {
+                return true;
+            }
+
+            //check parent of all nodes that are not DataCondition entries
+            var parent = GetUpperMostParent(dataDefinition, nodeParentDataDescription);
+
+            if (parentInError == null)
+            {
+                parentInError = new List<Node>();
+            }
+
+            if (parentInError.Contains(parent))
+            {
+                return true;
+            }
+
+            if (dataDefinition.CodeElement.Type == CodeElementType.DataConditionEntry ||
+                nodeParentDataDescription.Picture == null && (nodeParentDataDescription.Usage == null || 
+                                                              nodeParentDataDescription.Usage.Value == DataUsage.Display ||
+                                                              nodeParentDataDescription.Usage.Value == DataUsage.Index ||
+                                                              nodeParentDataDescription.Usage.Token.TokenType == TokenType.COMPUTATIONAL ) &&
```

Comment:
I made a mistake in specifications.
I've added a comment in #729 which list all cases to check.